### DISCUSSION TITLE
How to change favicon

### DISCUSSION THREAD

**Discussion by ArianYN** [dotnet/AspNetCore.Docs#33395] [Q&A]
I cannot for the life of me change the favicon of my blazor page. No matter what i do, it stays as the normal blazor icon.

My favicon is located as favicon.png inside wwwroot.

I have deleted the original favicon.png. In App.razor i have changed the link to look like this
`<link rel="icon" type="image/png" href="favicon.png" />`

That did not work. I have tried the same in my page itself, that didnt work either. Whenever i try to find solutions online, most of them just say the things i have tried, which does not work.

**Comment by NandanDevHub** [comment]
## Blazor WebAssembly

- **Edit `wwwroot/index.html`**  
  ```html
  <link rel="icon" type="image/x-icon" href="favicon.ico" />
  ```
  This tells the browser to load `favicon.ico` from your app’s `wwwroot` folder.

- **Place your `favicon.ico` in `wwwroot/`** (16×16 or 32×32).

## Blazor Server

- **In `Pages/_Host.cshtml` add this in `<head>`**  
  ```html
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  ```
  The leading `/` ensures ASP .NET Core’s static file middleware serves it from `wwwroot`.

- **Enable static files** by calling `app.UseStaticFiles();` before routing in `Program.cs` or `Startup.cs`.

## Clear Your Cache

- **Force reload** by appending a query string:  
  ```html
  <link rel="icon" href="favicon.ico?v=2" />
  ```
  Then do a hard refresh (Ctrl+F5) or open an incognito window.

**Comment by jumaevshahrom07** [comment]
One additional thing to check is whether the project contains more than one favicon declaration. The browser may still load an older `<link rel="icon">` entry.

Search the whole solution for:

```text
rel="icon"
favicon
```

Remove or update every old favicon reference.

For a Blazor Web App, place the new file in:

```text
wwwroot/site-icon.png
```

Then update the `<head>` section in `Components/App.razor`:

```html
<link rel="icon" type="image/png" href="/site-icon.png?v=3" />
```

Using a different filename such as `site-icon.png` is useful because browsers cache favicons aggressively.

You can verify that the file is served correctly by opening this address directly:

```text
https://localhost:PORT/site-icon.png?v=3
```

If the correct image opens:

1. Stop and restart the application.
2. Press `Ctrl + F5`.
3. Test in an incognito window.
4. Remove the old `favicon.ico` or old favicon link if it is still present.

Static assets should be stored under `wwwroot`, and a root-relative path such as `/site-icon.png` points to that folder.

**Reply by ArianYN** [reply]
Appreciate the answer, but this question is 2 years old lol
